### PR TITLE
HELM-96: Document the flow datasource.

### PR DESCRIPTION
**Pull request by j-white**
JIRA: https://issues.opennms.org/browse/HELM-96

### CODE REVIEW

File: docs/datasources/flow_datasource.adoc [hunk 1]
```diff
@@ -0,0 +1,63 @@
+= Flow Datasource
+
+== General
+
+The Flow Datasource can be used to retrieved application and conversation related metrics derived from flow data stored by OpenNMS.
```
Comment (mvrueden) [inline docs/datasources/flow_datasource.adoc]: _Flow Datasource_?

My english is not very good, but shouldn't it state `The _Flow Datasource_ can be used to retrieve application and conversation related metrics derived from flow data stored by OpenNMS`?

File: docs/datasources/flow_datasource.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+| Transform | `negativeIngress`       | Make the ingress values negative.
+| Transform | `asTableSummary`        | Generate a tabular summary instead of time series data.
+| Transform | `combineIngressEgress`  | Sum ingress and egress values together.
+|===
+
+== Template Queries
+
+=== exporterNodesWithFlows
+
+The `exporterNodesWithFlows()` query will return a list of nodes which have exported flow data to OpenNMS.
```
Comment (mvrueden) [inline docs/datasources/flow_datasource.adoc]: I guess the node must be known by OpenNMS, as exporting flows to OpenNMS is also possible if the exporter is not known by OpenNMS.

File: docs/datasources/flow_datasource.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+The value of the template variable is the `Node ID` and the display text is the `Node Label`.
+
+The available arguments are as follows:
+[options="header, autowidth"]
+|===
+| Name                 | Default       | Description
+| exportedWithinMillis | `14400000`    | Range of time in which a exporter must have a sent a flow to be included.
```
Comment (mvrueden) [inline docs/datasources/flow_datasource.adoc]: It seems there is an `a` to much here.

File: docs/datasources/flow_datasource.adoc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+=== interfacesOnExporterNodeWithFlows
+
+The `interfacesOnExporterNodeWithFlows()` query will return a list of interface indexes which are associated with one or more flows on the given node.
+The value of the template variable is the `Interface Index` and the display text is the `Interface Name`.
+
+The available arguments are as follows:
+[options="header, autowidth"]
+|===
+| Name                 | Default     | Description
+| nodeCriteria         | `(none)`    | Node criteria (Node ID, or Node Foreign Souce and Node Foreign ID) of the node of which to  enumerate the interfaces.
```
Comment (mvrueden) [inline docs/datasources/flow_datasource.adoc]: To us it is probably clear, what the values of `nodeCriteria` can be, but maybe an hint on the format is useful here?
E.g. `nodeCriteria: 5` vs `nodeCriteria: 'default:3242342342342'`

File: docs/datasources/flow_datasource.adoc (hunk 2 above)
Comment (j-white) [reply docs/datasources/flow_datasource.adoc]: I added a note bellow.

File: docs/datasources/flow_datasource.adoc (hunk 4 above)
Comment (j-white) [reply docs/datasources/flow_datasource.adoc]: I added some examples bellow.

**Comment by j-white** [comment]
Feedback was addressed. Merging.

**Pull request merged**